src/components/structures/MatrixChat.js

```diff
@@ -891,6 +895,7 @@ module.exports = React.createClass({
         });
 
         cli.on('sync', function(state, prevState) {
+            dis.dispatch({action: 'sync_state', prevState, state});
```

Comment:
The good thing about `flux.Store` is that it enforces the rule that everything that calls `__emitChange` (which is done after `_setState`) must be done during a dispatch. The js-sdk is not a dispatcher so `on.('sync', cb)` doesn't count and `cb` cannot cause the store to update.

So the store can't subscribe. As per my commit message, ideally there'd be a sync state store which registered as a listener and dispatched to itself a `sync_state` which would then propagate the actual sync state.